services/hanke-service/src/main/resources/db/changelog/changesets/add-tormaystarkastelu-results.yml

```diff
@@ -0,0 +1,29 @@
+databaseChangeLog:
+  - changeSet:
+      id: add-tormaystarkastelu-main-result-to-hanke-table
+      comment: Add main 'törmäystarkastelu' main result fields to hanke-table
+      author: Markku Hassinen
+      changes:
+        - addColumn:
+            tableName: hanke
+            columns:
+              - column: { name: liikennehaittaindeksi, type: float }
+              - column: { name: liikennehaittaindeksityyppi, type: varchar(30) }
+
+  - changeSet:
+      id: add-tormaystarkastelutulos-table
+      comment: Add 'törmäystarkastelu' results table
+      author: Markku Hassinen
+      changes:
+        - createTable:
+            tableName: tormaystarkastelutulos
+            columns:
+              - column: { name: id, type: int, autoIncrement: true, constraints: { primaryKey: true, nullable: false } }
+              - column: { name: hankeid, type: int, constraints: { nullable: false, foreignKeyName: fk_hanke_tormaystarkastelutulos, references: hanke(id) } }
+              - column: { name: liikennehaitta, type: float }
+              - column: { name: liikennehaittatyyppi, type: varchar(30) }
+              - column: { name: perus, type: float }
+              - column: { name: pyoraily, type: float }
+              - column: { name: joukkoliikenne, type: float }
+              - column: { name: tila, type: varchar(20) }
+              - column: { name: createdat, type: timestamp }
```

Comment:
Eli ei laiteta saraketta, että kuka teki? Miten valid-tilan muutos, eikö siitä tulisi modified-aikaleima?

Reply:
Ei ainakaan toistaiseksi; niin kauan kuin käyttäjä ei voi suoraan näitä arvoja manuaalisesti muuttaa (vaan ainoastaan lähdearvoja, joilla on oma entity ja omat audit-tiedot), niin modifiedBy olisi täällä aina "system". Ja koska lopputilannekin taisi olla toiveena että tältä osin kaikki olisi aina täysin automaattista, niin edes välivaiheessa jossa manuaalinen säätö ehkä olisi mahdollista, laittaisin ne tiedot vain normaali lokiin ("user 1234-bbbc-... modified törmäystarkastelutulos; old values ..., new values ..."). Se jättää myös historian; nämä nykyiset audit-kentät on muutenkin riittämättömät täyttämään lopulliset toiveet, ja saattavat siten jossain välissä tulevaisuudessa tippua pois koska "obsolete".

valid-tilan muutoksesta jälki onkin sitten mielenkiintoisempi. Nämä audit-kentät on tarkoitettu yleensä varsinaisten kenttien arvojen seurantaan, ei saman olion meta-kenttien (id, audit-kentät, apu-tilat).  Mutta toisaalta olisi oikein kiva tietää että milloin jokin tulos muuttui epäpäteväksi; onko siitä päivä vai vuosi... Tätä täytyy miettiä...

Reply:
Ai niin, ja laskennan laukaisemisesta pitäisi myös lokittaa (kuka/mikä ja milloin aloitti tormaystarkastelun).